### PR TITLE
Update Foreign Key Reference When Table Name Has Changed

### PR DESCRIPTION
**Pull request by s0kil**
Implements And Closes https://github.com/digitallyinduced/ihp/issues/748

### CODE REVIEW

File: IHP/IDE/SchemaDesigner/Controller/Columns.hs [hunk 1]
```diff
@@ -88,10 +88,28 @@ instance Controller ColumnsController where
...
+            Just constraint -> do
+                let constraintId = elemIndex constraint statements
+                case constraintId of
+                    Just constraintId -> do
+                        let constraintName = tableName <> "_ref_" <> columnName
+                        let referenceTable = get #referenceTable (get #constraint constraint)
+                        let Just onDelete = get #onDelete (get #constraint constraint)
+                        updateSchema (updateForeignKeyConstraint tableName columnName constraintName referenceTable onDelete constraintId)
+                    Nothing -> putStrLn ("Error")
+            Nothing -> pure ()
```
Comment (mpscholten) [inline IHP/IDE/SchemaDesigner/Controller/Columns.hs:111]: Could we extract this into it's own function? :) Then we should also add a test case for tihs.

**Comment by mpscholten** [comment]
Any updates?

**Comment by mpscholten** [comment]
@s0kil let's get this ready to merge :)

**Comment by s0kil** [comment]
Renaming table references is done, have to also do the columns.

File: IHP/IDE/SchemaDesigner/Controller/Columns.hs [hunk 2]
```diff
@@ -90,6 +90,27 @@ instance Controller ColumnsController where
...
+                --         let constraintId = elemIndex constraint statements
+                --         case constraintId of
+                --             Just constraintId -> do
+                --                 let constraintName = tableName <> "_ref_" <> columnName
+                --                 let referenceTable = get #referenceTable (get #constraint constraint)
+                --                 let Just onDelete = get #onDelete (get #constraint constraint)
+                --                 updateSchema (updateForeignKeyConstraint tableName columnName constraintName referenceTable onDelete constraintId)
+                --             Nothing -> putStrLn ("Error")
+                --     Nothing -> pure ()
+
```
Comment (mpscholten) [inline IHP/IDE/SchemaDesigner/Controller/Columns.hs:113]: Can this be removed?

Comment (s0kil) [reply IHP/IDE/SchemaDesigner/Controller/Columns.hs:113]: I'm not done with it yet, still have to rename columns.

**Comment by s0kil** [comment]
Maybe add a test case

**Review by mpscholten — APPROVED** [review]
Tested it locally and it works 👍

**Pull request merged**